packages/hub/src/vendor/hash-wasm/sha256-wrapper.ts

```diff
@@ -0,0 +1,63 @@
+import WasmModule from "./sha256";
+
+export async function createSHA256(isInsideWorker = false): Promise<{
+	init(): void;
+	update(data: Uint8Array): void;
+	digest(method: "hex"): string;
+}> {
+	const BUFFER_MAX_SIZE = 8 * 1024 * 1024;
+	const wasm: Awaited<ReturnType<typeof WasmModule>> = isInsideWorker
+		? // @ts-expect-error WasmModule will be populated inside self object
+		  await self["SHA256WasmModule"]()
+		: await WasmModule();
+	const heap = wasm.HEAPU8.subarray(wasm._GetBufferPtr());
+	return {
+		init() {
+			wasm._Hash_Init(256);
+		},
+		update(data: Uint8Array) {
+			let byteUsed = 0;
+			while (byteUsed < data.byteLength) {
+				const bytesLeft = data.byteLength - byteUsed;
+				const length = Math.min(bytesLeft, BUFFER_MAX_SIZE);
+				heap.set(data.subarray(byteUsed, length));
+				wasm._Hash_Update(length);
+				byteUsed += length;
+			}
+		},
+		digest(method: "hex") {
+			if (method !== "hex") {
+				throw new Error("Only digest hex is supported");
+			}
+			wasm._Hash_Final();
+			const result = Array.from(heap.slice(0, 32));
+			return result.map((b) => b.toString(16).padStart(2, "0")).join("");
+		},
+	};
+}
+
+export function createSHA256WorkerCode(): string {
+	return `
+    const _scriptDir = "";
```

Comment:
@coyotte508 Thanks for testing that out.

I had a look on the [minified version](https://cdn.jsdelivr.net/npm/@huggingface/hub@0.15.1-wasm/dist/browser/sha256-wrapper-X7SU6PWK.mjs/+esm). Turns out `_scriptDir` get optimized out.

I attempt to this fix in https://github.com/huggingface/huggingface.js/pull/682/commits/13df340932dd9c9ae96eef124c669614283d30a4 . Would you mind to redeploy it? Thanks!

Reply:
Just in case, you can do things like:

```ts
if (typeof _scriptDir !== undefined && _scriptDir) {
  ...
}
```

This should avoid failure when the variable is optimized out

Reply:
In fact, that's not really "optimized out" (Sorry my comment above was quite misleading - please ignore it).

What really happened is:
- emscripten set `_scriptDir` to `import.meta.url`, just to know where script is. This is useful for example when the `.js` and `.wasm` files are compiled separately, which is not our case.
- `_scriptDir` is provided at global scope of the module, so `WasmModule.toString()` cannot capture `_scriptDir`. That's why inside web worker code, I "fake" the value to empty string.

When the script is minified, `_scriptDir` is changed to another variable, that makes the "fake" value above no longer work. What I did in https://github.com/huggingface/huggingface.js/commit/13df340932dd9c9ae96eef124c669614283d30a4 was to use `sed` to patch that out inside the emscripten script.